src/Native/System.Net.Security.Native/pal_gssapi.cpp

```diff
@@ -0,0 +1,111 @@
+    return gss_display_status(minorStatus, statusValue, statusType, GSS_C_NO_OID, NULL, statusString);
+}
+
+extern "C" uint32_t GssImportName(uint32_t* minorStatus, char* inputName, bool isUser, gss_name_t* outputName)
+{
+    gss_buffer_desc inputNameBuffer {strlen(inputName), inputName};
+    gss_OID nameType = isUser ? const_cast<gss_OID>(GSS_C_NT_USER_NAME) : const_cast<gss_OID>(GSS_KRB5_NT_PRINCIPAL_NAME);
+    return gss_import_name(minorStatus, &inputNameBuffer, nameType, outputName);
+}
+
+extern "C" uint32_t GssInitSecContext(uint32_t* minorStatus, gss_cred_id_t claimantCredHandle, gss_ctx_id_t* contextHandle,
+                                            bool isNtlm, gss_name_t targetName, uint32_t reqFlags, uint8_t* inputBytes, int32_t inputLength, gss_buffer_t outputToken, uint32_t* retFlags)
+{
+#ifdef GSSFW
+    gss_OID desiredMech = isNtlm ? const_cast<gss_OID>(GSS_NTLM_MECHANISM) : gss_mech_spnego_OID;
+#else
+    assert(!isNtlm && "NTLM is not supported by MIT libgssapi_krb5");
+    gss_OID desiredMech = gss_mech_spnego_OID;
+#endif
+
+    gss_buffer_desc inputToken {static_cast<size_t>(inputLength), inputBytes};
+    return gss_init_sec_context(minorStatus, claimantCredHandle, contextHandle, targetName, desiredMech, reqFlags,
+                                0, GSS_C_NO_CHANNEL_BINDINGS, &inputToken, NULL, outputToken, retFlags, NULL);
+}
+
+extern "C" uint32_t GssInquireSourceName(uint32_t* minorStatus, gss_ctx_id_t contextHandle, gss_name_t* srcName)
+{
+    return gss_inquire_context(minorStatus, contextHandle, srcName, NULL, NULL, NULL, NULL, NULL, NULL);
+}
+
+extern "C" uint32_t GssReleaseCred(uint32_t* minorStatus, gss_cred_id_t* credHandle)
+{
+    return gss_release_cred(minorStatus, credHandle);
+}
+
+extern "C" uint32_t GssReleaseBuffer(uint32_t* minor_status, gss_buffer_t buffer)
+{
+    return gss_release_buffer(minor_status, buffer);
+}
+
+extern "C" uint32_t GssReleaseName(uint32_t* minorStatus, gss_name_t* inputName)
+{
+    return gss_release_name(minorStatus, inputName);
+}
+
+extern "C" uint32_t GssWrap(uint32_t* minorStatus, gss_ctx_id_t contextHandle, bool isEncrypt,
+                             uint8_t* inputBytes, int32_t offset, int32_t count, gss_buffer_t outputMessageBuffer)
+{
+    gss_buffer_desc inputMessageBuffer {static_cast<size_t>(count), inputBytes + offset};
+    int confState;
+    int confReqFlag = isEncrypt ? 1 : 0;
+    uint32_t retVal = gss_wrap(minorStatus, contextHandle, confReqFlag, GSS_C_QOP_DEFAULT, &inputMessageBuffer, &confState, outputMessageBuffer);
+    assert((confState == 0) == (confReqFlag == 0));
+    return retVal;
+}
+
+extern "C" uint32_t GssUnwrap(uint32_t* minorStatus, gss_ctx_id_t contextHandle,
+                             uint8_t* inputBytes, int32_t offset, int32_t count, gss_buffer_t outputMessageBuffer)
+{
+    gss_buffer_desc inputMessageBuffer {static_cast<size_t>(count), inputBytes + offset};
```

Comment:
Added a comment about that above - we have helper utilities for casting to and from size_t in Commong/pal_utilities.h